### DISCUSSION TITLE
kong upstream basic-authentication?

### DISCUSSION THREAD

**Discussion by tirelibirefe** [Kong/kong#12053] [Help] [answered]
Hello,
There are many resources regarding to key-auth or basic-auth for service and/or route. I am not asking to authenticate to a Kong service. 
I need to authenticate a service to an upstream.
![image](https://github.com/Kong/kong/assets/45227339/cfb62186-4362-48ba-8619-e2ee6c75f11b)

As far as I see, "basic-auth" plugin cannot set up a relation between consumer and route.

How can I accomplish the scenario above?

Thanks & Regards

**Comment by ms2008** [comment, marked as answer]
I think the easiest way to do this is to use the Kong Functions plugin and call the Kong PDK method `kong.service.request.set_header(header, value)` to set the `Authorization` header of the upstream in the access handler. That should make it work.

**Reply by tirelibirefe** [reply]
Thanks for the advise but how/where to use it?

**Reply by Water-Melon** [reply]
https://docs.konghq.com/hub/kong-inc/pre-function/how-to/#main
